soh/soh/Extractor/Extract.cpp

```diff
@@ -0,0 +1,535 @@
+            ShowSizeErrorBox();
+            continue;
+        }
+        inFile.read((char*)mRomData.get(), mCurRomSize);
+        RomToBigEndian(mRomData.get(), mCurRomSize);
+        verCrc = GetRomVerCrc();
+
+        // Rom doesn't claim to be valid
+        if (!verMap.contains(verCrc)) {
+            continue;
+        }
+
+        option = ShowRomPickBox(verCrc);
+        if (option == (int)ButtonId::YES) {
+            if (!ValidateRom(true)) {
+                if (rom == roms.back()) {
+                    ShowCrcErrorBox();
+                } else {
+                    ShowErrorBox("Rom CRC invalid",
+                                 "Rom CRC did not match the list of known good roms. Trying the next one...");
+                }
+                continue;
+            }
+            break;
+        } else if (option == (int)ButtonId::FIND) {
+            if (!GetRomPathFromBox()) {
+                ShowErrorBox("No rom selected", "No Rom selected. Exiting");
+                return false;
+            }
+            inFile.open(mCurrentRomPath, std::ios::in | std::ios::binary);
+            if (!inFile.is_open()) {
+                return false; // TODO Handle error
+            }
+            inFile.read((char*)mRomData.get(), mCurRomSize);
+            if (!ValidateRom()) {
+                return false;
+            }
+            break;
+        } else if (option == (int)ButtonId::NO) {
+            inFile.close();
+            if (rom == roms.back()) {
+                ShowErrorBox("No rom provided", "No rom provided. Exiting");
+                return false;
+            }
+            continue;
+        }
+        break;
+    }
+    return true;
+}
+
+bool Extractor::IsMasterQuest() const {
+    switch (GetRomVerCrc()) {
+        case OOT_PAL_GC_MQ_DBG:
+            return true;
+        case OOT_PAL_GC:
+        case OOT_PAL_GC_DBG1:
+            return false;
+        default:
+            UNREACHABLE;
+    }
+}
+
+const char* Extractor::GetZapdVerStr() const {
+    switch (GetRomVerCrc()) {
+        case OOT_PAL_GC:
+            return "GC_NMQ_PAL_F";
+        case OOT_PAL_GC_DBG1:
+            return "GC_NMQ_D";
+        case OOT_PAL_GC_MQ_DBG:
+            return "GC_MQ_D";
+        default:
+            // We should never be in a state where this path happens.
+            UNREACHABLE;
+            break;
+    }
+}
+
+#if 0
+const char* Extractor::GetZapdStr() {
+    constexpr size_t ZAPD_STR_SIZE = 1024;
+    char* zapdCall = new char[ZAPD_STR_SIZE];
+    const char* verStr = GetZapdVerStr();
+
+    // TODO anything would be better than this
+    snprintf(zapdCall, ZAPD_STR_SIZE,
+             "ed -i assets/extractor/xmls/%s -b %s -fl assets/extractor/filelists -o placeholder -osf placeholder -gsf "
+             "1 -rconf assets/extractor/Config_%s.xml -se OTR --otrfile %s",
+             verStr, mCurrentRomPath.c_str(), verStr, IsMasterQuest() ? "oot-mq.otr" : "oot.otr");
+
+    return zapdCall;
+}
+#endif
```

Comment:
looks like this is always if'd out, so unless we need it 
```suggestion
#endif
```